### PR TITLE
Set deployment timeout to "get some MB message" configurable

### PR DESCRIPTION
**Pull request by afjsilva**

### CODE REVIEW

File: src/Common/Resources.cs [hunk 1]
```diff
@@ -16,14 +16,15 @@ public class Resources
         public const string DEPLOYMENT_OUTPUTDIR_HELP = "Directory to place any artifacts generated by the deployment";
         public const string DEPLOYMENT_INTERACTIVE_HELP = "Flag that controls if the user should be prompted to go to the portal to initialize the installation manually";
         public const string DEPLOYMENT_TERMINATE_OTHER_VERSIONS_HELP = "Flag that controls if all the other versions of the Customer Environment should be terminated";
-        public const string DEPLOYMENT_TIMEOUT_MINUTES = "Number of minutes that are allowed to wait for the deployment to succeed. The default is 60 minutes.";
+        public const string DEPLOYMENT_TIMEOUT_MINUTES = "Timeout, in minutes, that are allowed to wait for the deployment to succeed. The default is 360 minutes (6 hours).";
+        public const string DEPLOYMENT_TIMEOUT_MINUTES_TO_GET_SOME_MB_MESSAGE = "Timeout, in minutes, that SDK client will wait to receive any message from portal by Message Bus. The default is 30 minutes.";
```
Comment (ritanorinho) [inline src/Common/Resources.cs:20]: "Timeout, in minutes, that the SDK client waits to receive any message from the portal via Message Bus. The default is 30 minutes."

File: src/Common/Services/AppInstallationHandler.cs [hunk 2]
```diff
@@ -29,9 +29,9 @@ public class AppInstallationHandler : IAppInstallationHandler
 
         private static DateTime? utcOfLastMessageReceived = null;
 
-        private TimeSpan timeoutMainTask = TimeSpan.FromMinutes(60);
+        private TimeSpan timeoutMainTask = TimeSpan.FromMinutes(360); // same timeout than RING (6 hours)
```
Comment (ritanorinho) [inline src/Common/Services/AppInstallationHandler.cs:32]: same timeout as ring

File: src/Console/DeployAgentCommand.cs [hunk 3]
```diff
@@ -22,6 +22,8 @@ public DeployAgentCommand(string name, string description = null) : base(name, d
         {
             Add(new Option<double?>(new[] { "--deploymentTimeoutMinutes", "-to", }, Resources.DEPLOYMENT_TIMEOUT_MINUTES));
 
+            Add(new Option<double?>(new[] { "--deploymentTimeoutMinutesToGetSomeMBMessage", "-tom", }, Resources.DEPLOYMENT_TIMEOUT_MINUTES_TO_GET_SOME_MB_MESSAGE));
```
[the remark this answers is not in the document]
Comment (afjsilva) [reply src/Console/DeployAgentCommand.cs:25]: timeoutmessage, that was the idea :)

File: src/Common/Services/AppInstallationHandler.cs [hunk 4]
```diff
@@ -229,7 +235,7 @@ public async Task Handle(string appName, CustomerEnvironmentApplicationPackage c
                                 compositeTokenSource.Dispose();
                                 cancellationTokenMBMessageReceived.Dispose();
 
-                                throw new TaskCanceledException($"Installation Failed! The installation timed out after {timeoutToGetSomeMBMessageTask.TotalMinutes} minutes without messages received on the MessageBus.");
+                                throw new TaskCanceledException($"Installation Failed! The installation timed out after {timeoutToGetSomeMBMessageTask.TotalMinutes} minutes because the SDK client did not receive additional expected messages on MessageBus from the portal and was waiting for installation to be finished.");
```
[the remark this answers is not in the document]
Comment (afjsilva) [reply src/Common/Services/AppInstallationHandler.cs:238]: please check again if it's better :)

File: src/Common/Services/EnvironmentDeploymentHandler.cs [hunk 5]
```diff
@@ -237,7 +243,7 @@ public async Task Handle(bool interactive, CustomerEnvironment customerEnvironme
                                 compositeTokenSource.Dispose();
                                 cancellationTokenMBMessageReceived.Dispose();
 
-                                throw new TaskCanceledException($"Deployment Failed! The deployment timed out after {timeoutToGetSomeMBMessageTask.TotalMinutes} minutes without messages received on MessageBus and waiting for deployment to be finished.");
+                                throw new TaskCanceledException($"Deployment Failed! The deployment timed out after {timeoutToGetSomeMBMessageTask.TotalMinutes} minutes because the SDK client did not receive additional expected messages on MessageBus from the portal and was waiting for deployment to be finished.");
```
[the remark this answers is not in the document]
Comment (afjsilva) [reply src/Common/Services/EnvironmentDeploymentHandler.cs:246]: same as above

File: src/Console/DeployAgentCommand.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (afjsilva) [reply src/Console/DeployAgentCommand.cs:25]: but I will change, please check again if it's better :)

File: src/Common/Services/AppInstallationHandler.cs [hunk 6]
```diff
@@ -29,9 +29,9 @@ public class AppInstallationHandler : IAppInstallationHandler
 
         private static DateTime? utcOfLastMessageReceived = null;
 
-        private TimeSpan timeoutMainTask = TimeSpan.FromMinutes(60);
+        private TimeSpan timeoutMainTask = TimeSpan.FromMinutes(360); // same timeout than RING (6 hours)
 
-        private TimeSpan timeoutToGetSomeMBMessageTask = TimeSpan.FromMinutes(15);
+        private TimeSpan timeoutToGetSomeMBMessageTask = TimeSpan.FromMinutes(30);
```
[the remark this answers is not in the document]
Comment (afjsilva) [reply src/Common/Services/AppInstallationHandler.cs]: Changed

**Review by ritanorinho — APPROVED** [review]

**Pull request merged**